### PR TITLE
TCN-875 Various bugs in release action

### PR DESCRIPTION
**Pull request by elliotmjackson**
1. contains extra `v` in pr output
2. does not trigger new tag after PR merge
3. does not create a new release on tag
4. logic to update `v1` branch never completed

all resolved.

### CODE REVIEW

File: .github/workflows/release.yaml [hunk 1]
```diff
@@ -35,14 +43,15 @@ jobs:
...
         with:
           add-paths: .
-          commit-message: "Update version to ${{env.VERSION}}"
-          branch: release/${{env.VERSION}}
+          commit-message: "Update version to v${{env.VERSION}}"
+          branch: release/v${{env.VERSION}}
           delete-branch: true
-          title: "Release ${{env.VERSION}}"
-          body: Release prepared for ${{env.VERSION}}
+          title: "Release v${{env.VERSION}}"
```
Comment (stefanvanburen) [inline .github/workflows/release.yaml:52]: won't these `v`s be doubled? (I realize this looks a lot like https://github.com/bufbuild/buf/blob/main/.github/workflows/release.yaml so maybe not?)

File: .github/workflows/release.yaml [hunk 2]
```diff
@@ -66,10 +75,39 @@ jobs:
...
+    steps:
+      - name: Generate token
+        id: generate_token
+        uses: tibdex/github-app-token@v1
+        with:
+          app_id: ${{env.APP_ID}}
+          private_key: ${{ secrets.TOKEN_EXCHANGE_GH_APP_PRIVATE_KEY }}
+          repository: ${{ github.repository }}
+          permissions: >-
+            {"contents": "write"}
```
Comment (stefanvanburen) [inline .github/workflows/release.yaml:97]: does this need `"pull_requests": "write"` permission, similar to [`buf`'s release workflow](https://github.com/bufbuild/buf/blob/7beb1e30eb91a52fb54f107f3a47515c5ad5168a/.github/workflows/release.yaml#L126)?

**Review by stefanvanburen — COMMENTED** [review]
I might not be the best person to review this one 😄; is this referencing release workflows we have elsewhere?

File: .github/workflows/release.yaml (hunk 2 above)
Comment (elliotmjackson) [reply .github/workflows/release.yaml:97]: this one does not, the one you reference is OP

File: .github/workflows/release.yaml (hunk 1 above)
Comment (elliotmjackson) [reply .github/workflows/release.yaml:52]: these will not, the input should be x.y.z without any v, even though that is not enforced at this stage 😂

**Review by stefanvanburen — APPROVED** [review]
stamping assuming you know what you're doing 😄

**Pull request merged**